### ISSUE TITLE
Not a hashmap

### ISSUE THREAD

**Issue by jackmoxley** [flesler/hashmap#47]
I just read the code, and you are literally just using a string as a key to an object property as your storage, your hash function doesn't even return a hash, it returns a string. 

Where are the buckets? 
Where is the correct hashing algorithm?

**Comment by flesler** [comment]
Hi, thanks for the feedback. I'm not interesting in theoretical debates. I'm closing the issue :+1:

**Comment by jackmoxley** [comment]
It's not theoretical, it's fact. It is not a hashmap.

**Comment by jackmoxley** [comment]
[Hash Map on Wikipedia](https://en.wikipedia.org/wiki/Hash_table)
Here is a simple explanation of how hashmaps work.

**Comment by jackmoxley** [comment]
@flesler https://github.com/jackmoxley/hashmap
I have implemented a hashmap that passes all your tests, apart from the hash ones, as they aren't hashing, and because hashmaps have no guarenteed order I have had to add sort in 3 places in the tests. Also it still supports NaN although I would argue that since NaN by JS spec shouldn't even equal itself, its a weird thing to want to support in a map. If we can remove it, it would simplify things down massively.
